### PR TITLE
Document how to mix in pregenerated skills dataset

### PR DESCRIPTION
**Pull request by bbrowning**
This documents how a user can mix in the pregenerated InstructLab community dataset for skills training. It's a bit manual at the moment, but until we get a nicer user experience this at least shares how to do it.

### CODE REVIEW

File: docs/data_mixing.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+The primary intended use of this is to specify an optional pregenerated dataset maintained by the InstructLab community that can improve training results when attempting to teach new skills to a model. This process is a bit manual for now, and the steps to do that are documented below.
+
+## Using InstructLab Community Pregenerated Dataset
+
+To use the [InstructLab Community pregenerated dataset](https://huggingface.co/datasets/instructlab/InstructLabCommunity) with all skills training, we first need to create a default recipe that specifies this dataset to include when mixing generated skills data. This recipe will get automatically picked up if placed in a `default_data_recipes/skills.yaml` subfolder and file under one of several possible locations - `'/home/<user>/.local/share/instructlab/sdg'`, `'/usr/local/share/instructlab/sdg'`, or `'/usr/share/instructlab/sdg'`. The exact list of possible locations is platform-dependent, and can be enumerated by a Python command like below:
```
Comment (markmc) [inline docs/data_mixing.md:9]: It's sufficient to just say the file is at `$XDG_DATA_DIRS/default_data_recipes/skills.yaml` as per https://specifications.freedesktop.org/basedir-spec/latest/#referencing

**Review by markmc — APPROVED** [review]
Nice. Thanks, Ben

**Review by khaledsulayman — COMMENTED** [review]
might be worth adding a section explaining to users how to write their own recipe entries. Could be showing an example so they see what it looks like in the yaml, or just briefly documenting `path`, `num_samples`, and `sampling_ratio`. WDYT?

**Review by aakankshaduggal — APPROVED** [review]
Thanks @bbrowning

**Comment by markmc** [comment]
> might be worth adding a section explaining to users how to write their own recipe entries. Could be showing an example so they see what it looks like in the yaml, or just briefly documenting `path`, `num_samples`, and `sampling_ratio`. WDYT?

That's a bit beyond what Ben was trying to document here - simply how to use the pregenerated InstructLab community dataset

**Pull request merged**